datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -0,0 +1,33 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.assignment;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.NotImplementedException;
+
+import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.DatastreamTask;
+
+
+/**
+ * Performs partition assignment based on partition throughput information
+ */
+public class LoadBasedPartitionAssigner {
```

Comment:
Shouldn't this be a pluggable interface ?

Reply:
We had this discussion with Vaibhav as well. For now we'll keep it as a class. When we have multiple implementations, we'll make it an interface.